network/junos/junos_config.py

```diff
@@ -32,6 +32,13 @@
     configuration and only changes are pushed to the device.
 extends_documentation_fragment: junos
 options:
+  mode:
+    description:
+      - By default you'll work in a shared candidate configuration. The 
+        C(mode) argument can be set to either private, dynamic, batch or
+        exclusive.
+    required: false
```

Comment:
And remove the list of choices that bit from the `description` as when you add in `choices:` That will get them in the documentation.

See http://docs.ansible.com/ansible/lineinfile_module.html for examples of `choices` in the documentation.